relay-server/src/services/project.rs

```diff
@@ -1164,18 +1192,26 @@ impl Project {
                 Outcome::RateLimited(reason_code),
             );
 
-            return None;
+            return Err(CheckBucketsError::RateLimited(len));
         }
 
         let project_metrics = ProjectMetrics {
             buckets,
             project_state,
         };
 
-        Some((scoping, project_metrics))
+        Ok((scoping, project_metrics))
     }
 }
 
+#[derive(Debug)]
```

Comment:
should this be actually a real error? 

You could derive `thiserror::Error` and add the proper display error for logging and wouldn't have to `map_err` but just log it once it's the `Err(err)`

Something like maybe ?
```rust
#[derive(Debug, thiserror::Error)]
enum CheckBucketsError {
    #[error("there is no project state: dropping {0} buckets")]
    ProjectExpired(usize),
...
```

Reply:
Yeah, that also works.
 
Could you, please, just add few lines of docs here?